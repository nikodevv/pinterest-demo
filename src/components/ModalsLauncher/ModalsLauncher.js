import React from 'react';
import * as redux from "react-redux";
import {Register} from "../Register/Register";
import NewLink from "../NewLink";
import CopyLink from "../CopyLink";


const ModalsLauncher = () => {
  const modals = redux.useSelector( state => state.modals);
  // The register component is a "special" modal that depends on the state of user's auth as opposed to state.modals.
  // It contains its own logic for whether it should be rendered or not.
  if (modals.showRegisterModal === true) {
    return <Register/>
  }
  else if (modals.showNewLinkModal === true) {
    return <NewLink/>
  }
  else if (modals.showShortLinkModal === true){
    return <CopyLink link={modals.shortLink}/>
  }
  return null;
};

export default ModalsLauncher;
