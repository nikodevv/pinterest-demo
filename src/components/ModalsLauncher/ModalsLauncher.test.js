import React from 'react';
import {configure, shallow} from 'enzyme';
import Adapter from "enzyme-adapter-react-16/build";
import {Provider} from "react-redux";
import configureMockStore from 'redux-mock-store';
import ModalsLauncher from './ModalsLauncher';


configure({adapter: new Adapter()});

const mockStore = configureMockStore();

describe('<ModalsLauncher />', () => {
  test('renders', () => {
    const wrapper = shallow(
      <Provider store={mockStore({})}>
        <ModalsLauncher />
      </Provider>);
    expect(wrapper).toMatchSnapshot();
  });
});
