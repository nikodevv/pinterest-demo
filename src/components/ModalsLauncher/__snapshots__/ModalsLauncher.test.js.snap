// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalsLauncher /> renders 1`] = `ShallowWrapper {}`;
